### PR TITLE
New(ci-operator): Red Hat Integration - Service Registry

### PR DESCRIPTION
**Pull request by rkubis**

### CODE REVIEW

**Comment by Sau1506mya** [comment]
Hi @rkubis , please mark this PR as WIP (Work in Progress) and then run `make update` command to generate the periodic job.

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
Waiting for merge of https://github.com/openshift/release/pull/38953

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by Sau1506mya** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Review by Simranpal — COMMENTED** [review]
Looks good so far. I was unsure about 2 things:
1. Presubmit job is missing
2. You haven't any env variables including the KUBECONFIG. Is this not needed for tests?

File: ci-operator/jobs/Apicurio/apicurio-registry-system-tests/OWNERS
Comment (ascerra) [inline ci-operator/jobs/Apicurio/apicurio-registry-system-tests/OWNERS:2]: Is there a reason that we left `rkubis` out of the approvers section here? I think it would be nice to allow more then just the cspi-qe team to approve.

File: ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml
Comment (ascerra) [inline ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml:27]:
```suggestion
  cron: 0 6 * * 1
  remote_api: true
```

Please add `remote_api: true` here as this will allow for us to trigger this job outside of the cron if needed.

**Comment by ascerra** [comment]
> Looks good so far. I was unsure about 2 things:
> 
> 1. Presubmit job is missing
> 2. You haven't any env variables including the KUBECONFIG. Is this not needed for tests?

@Simranpal I don't believe a presubmit job is needed here. We see presubmit jobs when we are building images based on a repos Dockerfile but in this case they are using an external Quay image

File: ci-operator/jobs/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main-periodics.yaml
Comment (ascerra) [inline ci-operator/jobs/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main-periodics.yaml:20]:
```suggestion
  name: periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3
  prowjob_defaults:
    tenant_id: gangway-api
```

Please add this here as this will allow for us to trigger this job outside of the cron if needed.

File: ci-operator/step-registry/service-registry/run-tests/service-registry-run-tests-commands.sh
Comment (ascerra) [inline ci-operator/step-registry/service-registry/run-tests/service-registry-run-tests-commands.sh:11]: We need to also make sure that the XML files start with `junit_`. This is needed for the test cases to appear on the TestGrid Dashboard for this scenario.

The XMLs generated are [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38858/rehearse-38858-periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3/1658725554989305856/artifacts/service-registry-2-3/service-registry-run-tests/artifacts/) these just need to read for example.. `junit_TEST-io.apicurio.registry.systemtests.api.BundleAPITests.xml`

using something like this should work

```
rename TEST junit_TEST /logs/artifacts/TEST*.xml
```

**Comment by ascerra** [comment]
With the release of OCP 4.13 today and this PR being created for the interop effort of testing against pre-release OCP, I believe we will need to update this to run on 4.14 prior to merge. The interop team will be discussing this further tomorrow morning and I will follow up with more guidance then.

File: ci-operator/jobs/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main-periodics.yaml
Comment (calebevans) [reply ci-operator/jobs/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main-periodics.yaml:20]: If you add the line @ascerra [mentioned earlier](https://github.com/openshift/release/pull/38858/files#r1196983286) and run `make update`, this will be done for you.

File: ci-operator/step-registry/service-registry/run-tests/service-registry-run-tests-commands.sh
Comment (calebevans) [reply ci-operator/step-registry/service-registry/run-tests/service-registry-run-tests-commands.sh:11]: This will also be required to report [Jira bugs on test failures when the project is done](https://issues.redhat.com/browse/LPTOCPCI-138).

**Review by calebevans — CHANGES_REQUESTED** [review]
I didn't see anything other than what @ascerra has already pointed out.

File: ci-operator/jobs/Apicurio/apicurio-registry-system-tests/OWNERS
Comment (Sau1506mya) [reply ci-operator/jobs/Apicurio/apicurio-registry-system-tests/OWNERS:2]: Okay, I will add @rkubis also as an approver here

**Comment by rkubis** [comment]
> Looks good so far. I was unsure about 2 things:
> 
>     1. Presubmit job is missing
> 
>     2. You haven't any env variables including the KUBECONFIG. Is this not needed for tests?

Ad 2. All configuration for tests is constant, there is not need to have any environment variables.

**Comment by rkubis** [comment]
> With the release of OCP 4.13 today and this PR being created for the interop effort of testing against pre-release OCP, I believe we will need to update this to run on 4.14 prior to merge. The interop team will be discussing this further tomorrow morning and I will follow up with more guidance then.

It looks so, thanks!

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.13-lp-interop-service-registry-2-3

File: ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml
Comment (ascerra) [inline ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml:19]:
```suggestion
      version: "4.14"
```

File: ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml
Comment (ascerra) [inline ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml:3]:
```suggestion
    name: "4.14"
```

File: ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml
Comment (ascerra) [inline ci-operator/config/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main__service-registry-ocp4.13-lp-interop.yaml:41]: Change 4.13 to 4.14 in the filename of this config. Then once you run `make update` after that this metadata will be updated.

**Comment by rkubis** [comment]
/pj-rehearse periodic-ci-Apicurio-apicurio-registry-system-tests-main-service-registry-ocp4.14-lp-interop-service-registry-2-3

File: ci-operator/jobs/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main-periodics.yaml
Comment (chetna14manku) [inline ci-operator/jobs/Apicurio/apicurio-registry-system-tests/Apicurio-apicurio-registry-system-tests-main-periodics.yaml:25]: Is this a private channel? I am unable to find it on slack?

**Review by calebevans — APPROVED** [review]
/lgtm

@Sau1506mya you'll need to add a pj-rehearse acknowledgement and remove "WIP" from the title for this to merge.

**Pull request merged**
